src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1164,6 +1168,32 @@ void OrbitMainWindow::LoadCaptureOptionsIntoApp() {
   app_->SetTraceGpuSubmissions(settings.value(kTraceGpuSubmissionsSettingKey, true).toBool());
   app_->SetEnableApi(settings.value(kEnableApiSettingKey, true).toBool());
   app_->SetEnableIntrospection(settings.value(kEnableIntrospectionSettingKey, false).toBool());
+  bool const collect_callstack_on_thread_state_change =
+      settings
+          .value(kEnableCallStackCollectionOnThreadStateChanges,
+                 orbit_qt::CaptureOptionsDialog::kThreadStateChangeCallStackCollectionDefaultValue)
+          .toBool();
+  CaptureOptions::UnwindingMethod const tracepoint_callstack_method =
+      static_cast<CaptureOptions::UnwindingMethod>(
+          settings
+              .value(kCallStackCollectionOnThreadStateChangesMethod,
+                     orbit_qt::CaptureOptionsDialog::kThreadStateChangeCallStackMethodDefaultValue)
+              .toInt());
+  if (settings.value(kCollectThreadStatesSettingKey, false).toBool()) {
+    if (!collect_callstack_on_thread_state_change) {
+      app_->SetThreadStateChangeCallstackCollection(
+          CaptureOptions::kNoThreadStateChangeCallStackCollection);
+      app_->SetThreadStateChangeCallstackMethod(orbit_grpc_protos::CaptureOptions::kUndefined);
```

Comment:
I would expect that this is automatically set by the line above (setting to `kNoThreadStateChangeCallStackCollection`) or either that setting to Undefined is not needed anyways.

Reply:
Setting to Undefined is not needed anyways but I did it anyway, it won't affect anything.